src/keybindings/keybindings.rs

```diff
@@ -0,0 +1,121 @@
+use serde::Deserialize;
+
+use crate::opts::KeybindingsSection;
+
+use super::{action::Action, KeyCombo};
+
+/// A list of [`keybindings`](KeyCombo) each associated with an [`Action`].
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+pub struct Keybindings(Vec<(Action, KeyCombo)>);
+
+impl Keybindings {
+    /// Returns an iterator over the [`Action`]s and [`KeyCombo`]s
+    pub fn iter(&self) -> std::slice::Iter<'_, (Action, KeyCombo)> {
+        self.0.iter()
+    }
+}
+
+impl Extend<(Action, KeyCombo)> for Keybindings {
+    fn extend<I: IntoIterator<Item = (Action, KeyCombo)>>(&mut self, iter: I) {
+        self.0.extend(iter)
+    }
+}
+
+impl IntoIterator for Keybindings {
+    type Item = (Action, KeyCombo);
+    type IntoIter = <Vec<(Action, KeyCombo)> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl Default for Keybindings {
+    fn default() -> Self {
+        Self(super::defaults::defaults())
+    }
+}
+
+impl From<KeybindingsSection> for Keybindings {
+    /// Converts from [`KeybindingsSection`] to [`Keybindings`].
+    ///
+    /// If an `extra` keybinding collides with a `base` one, then the `base` one is dropped in
+    /// favor of the `extra` keybinding
+    fn from(value: KeybindingsSection) -> Self {
+        let mut base = value.base;
+
+        if let Some(extra) = value.extra {
+            for (_, extra_combo) in extra.iter() {
+                base.0 = base
+                    .clone()
+                    .into_iter()
+                    .filter(|(_, combo)| !combo.starts_with(extra_combo))
+                    .collect();
+            }
```

Comment:
This is the relevant part of the PR, we eliminate `base` bindings that appear in `extra`

Reply:
Probably good to log the combos that are getting filtered to make debugging easier

Reply:
I agree, I'll add that